lib/gpu_memory_service/integrations/vllm/model_loader.py

```
@@ -103,13 +161,26 @@ def _load_read_mode(
 
     try:
         model = _create_meta_model(vllm_config, model_config)
         materialize_module_from_gms(gms_client, model, device_index=device_index)
 
+        # MX: register materialized tensors (available for P2P transfer)
+        mx_ctx = get_mx_load_context(vllm_config, model_config)
+        if mx_ctx is not None:
+            try:
+                register_tensors(model, mx_ctx)
+                publish_metadata(mx_ctx)
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)